plugins/state_history_plugin/session.hpp

```diff
@@ -0,0 +1,421 @@
+#pragma once
+#include <eosio/chain/block_state.hpp>
+#include <eosio/state_history/compression.hpp>
+#include <eosio/state_history/log.hpp>
+#include <eosio/state_history/serialization.hpp>
+
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/local/stream_protocol.hpp>
+#include <boost/beast/websocket.hpp>
+
+
+extern const char* const state_history_plugin_abi;
+
+namespace eosio {
+using namespace state_history;
+using tcp     = boost::asio::ip::tcp;
+using unixs   = boost::asio::local::stream_protocol;
+namespace ws  = boost::beast::websocket;
+namespace bio = boost::iostreams;
+
+template <typename Session>
+struct send_queue_entry_base {
+   virtual ~send_queue_entry_base() {}
+   virtual void send(Session* s) = 0;
+};
+
+template <typename Session>
+struct basic_send_queue_entry : send_queue_entry_base<Session> {
+   std::vector<char> data;
+   template <typename... Args>
+   basic_send_queue_entry(Args&&... args)
+       : data(std::forward<Args>(args)...) {}
+   void send(Session* s) override {
+      s->socket_stream.async_write(boost::asio::buffer(data),
+                                   [s = s->shared_from_this()](boost::system::error_code ec, size_t) {
+                                      s->callback(ec, "async_write", [s] { s->pop_entry(); });
+                                   });
+   }
+};
+
+template <typename Session>
+class blocks_result_send_queue_entry : public send_queue_entry_base<Session> {
+   eosio::state_history::get_blocks_result_v0                    r;
+   std::vector<char>                                             data;
+   std::variant<std::vector<char>, maybe_locked_compress_stream> buf;
+
+   template <typename Next>
+   void async_send(Session* s, bool fin, const std::vector<char>& data, Next&& next) {
```

Comment:
The param names of `data` is very confusing with the member `data`. Maybe use `d` here instead.